### PR TITLE
add visualization_data

### PR DESCRIPTION
**Pull request by jw3126**

### CODE REVIEW

File: test/VisualizationTests/VisualizationDataTests.jl [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  conformity=:H1, model=model)
+U = TrialFESpace(V0)
+trian = Triangulation(model)
+degree = 2
+quad = CellQuadrature(trian,degree)
+A(u, v) = u ⊙ v
+b(v) = v⊙f
+t_Ω = AffineFETerm(A,b,trian,quad)
+op = AffineFEOperator(U,V0,t_Ω)
+u = solve(op)
```
Comment (jw3126) [inline test/VisualizationTests/VisualizationDataTests.jl:22]: What is a simpler way to create fields then solving a FEM problem?

File: test/VisualizationTests/VisualizationDataTests.jl [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+visdata = visualization_data(trian, cellfields=Dict("u" =>u))
+ncells = 12
+nodes_per_cell = 3
```
Comment (jw3126) [inline test/VisualizationTests/VisualizationDataTests.jl:26]: How to extract these numbers directly from `trian` or `model`?

File: src/Visualization/VisualizationData.jl [hunk 3]
```diff
@@ -0,0 +1,25 @@
+"""
+"""
+function visualization_data(trian::Triangulation; order=-1, nsubcells=-1, celldata=Dict(), cellfields=Dict())
```
Comment (jw3126) [inline src/Visualization/VisualizationData.jl:3]: Should we mark this function experimental? I think we might want to change it in future. Like returning a `struct` instead of a named tuple etc.

File: test/VisualizationTests/VisualizationDataTests.jl (hunk 1 above)
Comment (fverdugo) [reply test/VisualizationTests/VisualizationDataTests.jl:22]: By interpolation
```julia
f(x) = sin(4*x[1])*cos(3*x[2])
u = interpolate(U,f)
```

In this case U can even be a test space without boundary conditions. You don't need to build a trial one.

Comment (jw3126) [reply test/VisualizationTests/VisualizationDataTests.jl:22]: perfect thanks!

File: test/VisualizationTests/VisualizationDataTests.jl (hunk 2 above)
Comment (fverdugo) [reply test/VisualizationTests/VisualizationDataTests.jl:26]:
```julia
using Gridap.Geometry
using Gridap.ReferenceFEs
ncells = num_cells(vistrian)
nodes_per_cell = num_nodes(first(get_reffes(vistrian)))
```

File: src/Visualization/VisualizationData.jl (hunk 3 above)
Comment (fverdugo) [reply src/Visualization/VisualizationData.jl:3]: Don't worry about this.

File: test/VisualizationTests/VisualizationDataTests.jl (hunk 1 above)
Comment (jw3126) [reply test/VisualizationTests/VisualizationDataTests.jl:22]: Btw. it would be nice if the argument order of `interpolate` was switched. The usual convention is functions first, as it enables do block syntax.

Comment (fverdugo) [reply test/VisualizationTests/VisualizationDataTests.jl:24]: > Btw. it would be nice if the argument order of interpolate was switched. The usual convention is functions first, as it enables do block syntax.

true.

**Comment by jw3126** [comment]
I added `VisualizationData` struct. I think we will need to dispatch on something like `AbstractPlotting.plot!(::Plot(VisualizationData)` later.

**Comment by jw3126** [comment]
@fverdugo can you merge this and add me to GridapMakie? I have some code I would like to put there.

**Comment by fverdugo** [comment]
Hi @jw3126,

I would like to discuss what we want to put in GridapMakie before granting access. We can discuss it tomorrow e.g. via Gitter
